ksql-engine/src/test/resources/query-validation-tests/key-field.json

```diff
@@ -628,22 +718,28 @@
         {"topic": "OUTPUT", "key":"2", "value": {"ALIASED":2, "KSQL_COL_1": 1}}
       ],
       "post": {
-        "issues": [
-          "key field of output has INCORRECT name: should be 'ALIASED' - (see Issue #1 above)",
-          "key field of output has INCORRECT type: should be 'INT' - (see Issue #3 above)"
-        ],
         "sources": [
-          {"name": "OUTPUT", "type": "table", "keyField": {"name": "KSQL_INTERNAL_COL_0", "schema": {"type": "STRING"}}}
+          {"name": "OUTPUT", "type": "table", "keyField": {"name": "ALIASED", "legacyName": "KSQL_INTERNAL_COL_0", "legacySchema": {"type": "STRING"}}}
```

Comment:
this behavior is a little confusing to me (I know the "should" parts of it were in a previous PR, but I've been able to digest a little more):
```sql
0: CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo');
1: CREATE TABLE OUTPUT AS SELECT bar AS aliased, COUNT(*) FROM INPUT GROUP BY bar;
2: CREATE TABLE OUTPUT AS SELECT foo AS bar, COUNT(*) FROM INPUT GROUP BY bar;
```
It's a little hard for me to grep the test file to see if a test for `2` exists, but what does it do? Does it key as the old `bar` or the new `bar` (i.e. the old `foo`). I feel like grouping by the old `bar` would throw me off, but that contradicts what `1` does...

In my opinion, we should fail a `GROUP BY` if it uses the old name of an aliased field.

Reply:
Statement no 2. would fail to parse. It would complain 'bar' is not known.  (Again, existing functionality).

Inconsistently, `PARTITION BY` would work for statement no. 2, but fail with a similar error for statement no. 1.  But that's a problem for another day.

Reply:
Got it - it makes more sense that way (it's always in the source, doesn't matter what the alias in the result is).